### PR TITLE
Cleanup of stepper.cpp

### PR DESCRIPTION
**Pull request by thinkyhead**
- Rename `WRITE_E_STEP` for consistency, use in auto-name generation
- Add `BIT` and `TEST` macros to make bits and tests semantic
- Add `HAS_DIGIPOTSS` macro to reduce redundancy
- Add `*_APPLY_*` macros to `stepper.cpp` to help consolidate movement code, including dual axes
- Consolidate repetitive code in `stepper.cpp` using macros
- Apply coding standards in `stepper.cpp`
- Use `>= 0` instead of `> -1` to indicate "0" is the floor
- Replace legacy `DUAL_Y_CARRIAGE` with `Y_DUAL_STEPPER_DRIVERS` (#1604)

### CODE REVIEW

**Comment by alexborro** [comment]
I think this is just a code clean up, right?

Please take additional care when touching stepper.cpp. This is the second
mass change in this file this week.

Cheers.

Alex.
Em 14/03/2015 08:37, "Scott Lahteine" notifications@github.com escreveu:

**Comment by thinkyhead** [comment]
@alexborro Oh yes, I am taking the most meticulous care, and _studying_ the code carefully as I go along. It was good to go through it with a fine-toothed comb and get to know this core element. It is pretty optimal, so I didn't try to save cycles, just reduce redundancy and improve consistency.

**Comment by thinkyhead** [comment]
More combing, and now checking out how C++ optimizes by looking at the .s assembly output. I used to program 6502 and then 680x0 assembly language, so it's all so weirdly familiar… but so many more bits and registers…

File: Marlin/SdBaseFile.h [hunk 1]
```diff
@@ -171,9 +171,9 @@ static inline uint8_t FAT_SECOND(uint16_t fatTime) {
   return 2*(fatTime & 0X1F);
 }
 /** Default date for file timestamps is 1 Jan 2000 */
-uint16_t const FAT_DEFAULT_DATE = ((2000 - 1980) << 9) | (1 << 5) | 1;
+uint16_t const FAT_DEFAULT_DATE = ((2000 - 1980) << 9) | BIT(5) | 1;
```
Comment (ledvinap) [inline Marlin/SdBaseFile.h:174]: (1<<5) is not always BIT(5) ...

**Comment by thinkyhead** [comment]
@ledvinap How so? And in any case, the BIT macro is defined as

```
#define BIT(b) (1<<(b))
```

so it is literally the same code.

**Comment by ledvinap** [comment]
Yes, technically the code is identical.

The meaning is `(year << 9) | (month << 5)|(day << 0);`. Your change makes no sense and will confuse everyone reading the code in future.

**Comment by thinkyhead** [comment]
@ledvinap Ah, I see. More an issue of semantic consistency. Yes, that will be reverted.

**Pull request merged**